modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/service/test/SegmentsExperimentLocalServiceTest.java

```diff
@@ -674,43 +688,140 @@ public void testUpdateSegmentsExperimentNameInStatusRunning()
+		SegmentsExperience segmentsExperience3 = _addSegmentsExperience(-3);
 
 		SegmentsExperiment segmentsExperiment = _addSegmentsExperiment(
-			segmentsExperience1);
+			segmentsExperience2);
 
 		SegmentsExperience variantSegmentsExperience =
-			SegmentsTestUtil.addSegmentsExperience(
-				segmentsExperience1.getGroupId(),
-				segmentsExperience1.getClassNameId(),
-				segmentsExperience1.getClassPK());
+			_addVariantSegmentsExperience(
+				segmentsExperience2.getSegmentsEntryId(),
+				segmentsExperiment.getSegmentsExperimentId());
 
-		_segmentsExperienceLocalService.updateSegmentsExperienceActive(
-			variantSegmentsExperience.getSegmentsExperienceId(), false);
+		segmentsExperiment.setStatus(
+			SegmentsExperimentConstants.STATUS_FINISHED_WINNER);
 
-		_segmentsExperimentRelLocalService.addSegmentsExperimentRel(
-			segmentsExperiment.getSegmentsExperimentId(),
-			variantSegmentsExperience.getSegmentsExperienceId(),
-			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+		_segmentsExperimentLocalService.updateSegmentsExperiment(
+			segmentsExperiment);
 
 		_segmentsExperimentLocalService.updateSegmentsExperimentStatus(
 			segmentsExperiment.getSegmentsExperimentId(),
-			SegmentsExperimentConstants.STATUS_RUNNING);
+			variantSegmentsExperience.getSegmentsExperienceId(),
+			SegmentsExperimentConstants.STATUS_COMPLETED);
+
+		segmentsExperience1 =
+			_segmentsExperienceLocalService.fetchSegmentsExperience(
+				segmentsExperience1.getSegmentsExperienceId());
+
+		Assert.assertTrue(segmentsExperience1.isActive());
+		Assert.assertEquals(0, segmentsExperience1.getPriority());
+
+		segmentsExperience2 =
+			_segmentsExperienceLocalService.fetchSegmentsExperience(
+				segmentsExperience2.getSegmentsExperienceId());
+
+		Assert.assertFalse(segmentsExperience2.isActive());
+		Assert.assertEquals(1, segmentsExperience2.getPriority());
+
+		segmentsExperience3 =
+			_segmentsExperienceLocalService.fetchSegmentsExperience(
+				segmentsExperience3.getSegmentsExperienceId());
+
+		Assert.assertTrue(segmentsExperience3.isActive());
+		Assert.assertEquals(-3, segmentsExperience3.getPriority());
+
+		variantSegmentsExperience =
+			_segmentsExperienceLocalService.fetchSegmentsExperience(
+				variantSegmentsExperience.getSegmentsExperienceId());
+
+		Assert.assertTrue(variantSegmentsExperience.isActive());
+		Assert.assertEquals(-2, variantSegmentsExperience.getPriority());
+	}
+
+	@Test
+	public void testUpdateSegmentsExperimentStatusToCompletedWithWinnerSegmentsExperienceAndControlSegmentsExperienceWithPriorityZero()
+		throws Exception {
+
+		SegmentsExperience segmentsExperience1 = _addSegmentsExperience(1);
+		SegmentsExperience segmentsExperience2 = _addSegmentsExperience(0);
+		SegmentsExperience segmentsExperience3 = _addSegmentsExperience(-2);
+
+		SegmentsExperiment segmentsExperiment = _addSegmentsExperiment(
+			segmentsExperience2);
+
+		SegmentsExperience variantSegmentsExperience =
```

Comment:
Can't we add the variant as Winner directly?

Reply:
what do you mean? I mean, the variant has not a status is the Experiment. Do you mean to create an experiment directly in STATUS_FINISHED_WINNER?

Reply:
Yeahh

Reply:
I'm afraid It's not possible, unless we use directly the persistence layer. There's a transition status validation, and It's not possible to create an experiment directly to a status.  The first one is always DRAFT

Reply:
OK